### DISCUSSION TITLE
No Body-Text with Attachment

### DISCUSSION THREAD

**Discussion by saturnus69** [PHPMailer/PHPMailer#2722] [Q&A]
I use PHP 8.1.7 and PHPMailer 6.6.3.
As soon as I send an email with an attachment I get the email without body text and the attachment (PDF) cannot be opened. The path is correct, the PDF too. 

$mail->addAttachment('../stuff/uploads/pdf/1408_dossier.pdf','1408_dossier.pdf');

Without attachment I receive the e-mail correctly.
These problems do not occur when I use PHPMailer 6.1.8.
Does anyone know where the problem is?

**Comment by Synchro** [comment]
Can you share your code and an example of a received message? Is `addAttachment` returning true?

**Reply by saturnus69** [reply]
Yes, addAttachment is returning true.

`       $mail->IsHTML(true); 
	$mail->CharSet = 'utf-8';
	$mail->From = 'noreply@myfirma.ch';
	$mail->FromName = 'Firma - '.$_SERVER['HTTP_HOST'].'';	
	$mail->AddBCC('myemail@example.com');
        $mail->Subject = 'Reservation ' . $_SERVER['HTTP_HOST'];
	$mail->Body  = nl2br("This is the Message\r\n"); 
	$mail->Body .= nl2br(more Text..."\r\n");
        $mail->addAttachment('../stuff/uploads/pdf/1408_dossier.pdf','1408_dossier.pdf');
     
	if(!$mail->Send()){
           echo $mail->ErrorInfo;
           exit();
        }`

The received Message is empty with the attachment pdf (broken)

**Reply by Synchro** [reply]
Try a minimal example, like a text file containing a single character, and please show the full SMTP transcript with `SMTPDebug = 2`.

**Reply by saturnus69** [reply]
still the same with a txt file with only one single character. i don't send mails with smtp. What exactly does it mean "SMTP transcript with SMTPDebug=2"?

**Reply by Synchro** [reply]
OK, try this:

    $mail->isSMTP();
    $mail->SMTPDebug = 2;

It will show exactly what's being sent. It should work because by default the PHP mail function simply opens a synchronous SMTP connection to localhost anyway. Using the mail function can be difficult to debug because it doesn't let you know what's happening.

**Reply by saturnus69** [reply]
OK, now i get this:

2022-06-28 09:11:49 SERVER -> CLIENT: 220 domain.ch ESMTP
2022-06-28 09:11:49 CLIENT -> SERVER: EHLO domain.ch
2022-06-28 09:11:49 SERVER -> CLIENT: 250-domain.ch250-AUTH=LOGIN CRAM-MD5 PLAIN250-AUTH LOGIN CRAM-MD5 PLAIN250-STARTTLS250-PIPELINING250 8BITMIME
2022-06-28 09:11:49 CLIENT -> SERVER: STARTTLS
2022-06-28 09:11:49 SERVER -> CLIENT: 220 ready for tls
SMTP Error: Could not connect to SMTP host. Connection failed. stream_socket_enable_crypto(): Peer certificate CN=`domain.ch' did not match expected CN=`localhost'
2022-06-28 09:11:49 CLIENT -> SERVER: QUIT
2022-06-28 09:11:49
2022-06-28 09:11:49
SMTP Error: Could not connect to SMTP host. Connection failed. stream_socket_enable_crypto(): Peer certificate CN=`domain.ch' did not match expected CN=`localhost'

**Reply by Synchro** [reply]
Ah, ok, add this:

    $mail->SMTPAutoTLS = false;

**Reply by Synchro** [reply]
Your mail server is a little odd since it's asking for authentication on localhost, which is unusual. Since your PHP mail() calls apparently work, you should be able to do without auth though.

**Reply by saturnus69** [reply]
I don't know anything about this technically. Not much has changed.
SMTP Error: The following recipients failed: myemail@example.ch: sorry, that domain isn't in my list of allowed rcpthosts (#5.7.1)
Interestingly, it works with version 6.1.8....

**Reply by Synchro** [reply]
Can you show the **full** debug output with 6.1.8?

**Reply by saturnus69** [reply]
2022-06-28 10:13:50 SERVER -> CLIENT: 220 domain.ch ESMTP
2022-06-28 10:13:50 CLIENT -> SERVER: EHLO domain.ch
2022-06-28 10:13:50 SERVER -> CLIENT: 250-independence.raq.ch250-AUTH=LOGIN CRAM-MD5 PLAIN250-AUTH LOGIN CRAM-MD5 PLAIN250-STARTTLS250-PIPELINING250 8BITMIME
2022-06-28 10:13:50 CLIENT -> SERVER: MAIL FROM:<noreply@domain.ch>
2022-06-28 10:13:50 SERVER -> CLIENT: 250 ok
2022-06-28 10:13:50 CLIENT -> SERVER: RCPT TO:<myemail@example.ch>
2022-06-28 10:13:50 SERVER -> CLIENT: 553 sorry, that domain isn't in my list of allowed rcpthosts (#5.7.1)
2022-06-28 10:13:50 SMTP ERROR: RCPT TO command failed: 553 sorry, that domain isn't in my list of allowed rcpthosts (#5.7.1)
2022-06-28 10:13:50 CLIENT -> SERVER: QUIT
2022-06-28 10:13:50 SERVER -> CLIENT: 221 domain.ch
SMTP Error: The following recipients failed: myemail@example.ch: sorry, that domain isn't in my list of allowed rcpthosts (#5.7.1)

**Reply by Synchro** [reply]
So it doesn't work with 6.1.8 either?

Do you have some credentials you can put in, like this:

```php
$mail->Username = 'username';
$mail->Password = 'password';
$mail->SMTPAuth = true;
```

**Reply by saturnus69** [reply]
Yes, it works with this version, but not newer then 6.1.8. I dont have this informations at this time, sorry. 
think i keep the working version. I go home now. Thank you for your time and help. I appreciate that. :-)
